### DISCUSSION TITLE
2011-11-22 Core team update

### DISCUSSION THREAD

**Discussion by mm-near** [near/community#132] [Core]
# Highlights:
* new validation selection algorithm is launching next week on testnet

# TL;DR
@pmnoxx - improving network performance / network rate-limiting
@Longarithm - improving fees / debugging performance inconsistencies / splitting configs (Runtime and Genesis)
@mzhangmzz -   improve chunk fetching / improving block validation error handling
@yordanmadzhunkov - fuzz-testing for  'more validators' launch on Nov 29
@EgorKulikov - working on different validator assignment (for 'more validators' on Nov 29)
@abacabadabacaba - better storage analysis (replace Merkle tree with AVL)

# Done:
@pmnoxx - lots of network refactoring code (around 30 PRs) with documentation improvements
@pmnoxx - adding DeepSize to our objects, that we can use for rate limiting
@pmnoxx - adding benchmark for storage- to help @Longarithm debug the performance inconsistencies

@Longarithm - looked at how CosmWasm handles the fees
@Longarithm - removing Defaults from RuntimeConfig (near/nearcore#5292)
@Longarithm - removed unused max_gas_burnt_view field near/nearcore#5297
@Longarithm - estimated gas spent on funcall with enabled caching - it’s 5.5x in the best case
@Longarithm - found reason of only 48% successful get’s in DB benchmark (bug in our benchmarking code)

@mzhangmzz - Fixed a bug in chunks requesting so that non-validations will not wait for chunks being forwarded anymore near/nearcore#5360
@mzhangmzz - Worked on requesting chunks for multiple blocks simultaneously, added comments and tests to the PR near/nearcore#5224
@mzhangmzz - Debugged issues emerged in sharding upgrade on mainnet

@EgorKulikov - PR to split new validator selection algorithm from chunk only producers
@yordanmadzhunkov - Found last source of flakiness in block production (randomization of transaction pool)
@yordanmadzhunkov - started a PR to make it deterministic in tests.

@mm-near - fixed 3 nayduck tests

# Progress:

@pmnoxx - NearClock - a wrapper around SystemClock, which is compatible with existing Clock mocking (near/nearcore#5177)
@pmnoxx - reference level documentation for networking code (many PRs)

@Longarithm - compute economic effect of under/overcharging funcall/deployment fees
@Longarithm - propose solutions for new funcall/deployment fees
@Longarithm - play with DB parameters to solve perf cliff issue

@mzhangmzz - Take a look at our block validation code and try to restructure error types in process_block, see comment near/nearcore#5270 (comment)

@EgorKulikov - Merge & stabilize the validators PR

@yordanmadzhunkov - fuzz testing production code with coverage - focusing on the validator selection algorithm
